contracts/compressor/IntentCompressor.sol

```diff
@@ -0,0 +1,335 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IInbox} from "../interfaces/IInbox.sol";
+import {IIntentSource} from "../interfaces/IIntentSource.sol";
+import {Intent, Call, TokenAmount, Route, Reward} from "../types/Intent.sol";
+
+struct EncodedIntent {
+    uint8 sourceChainIndex;
+    uint8 destinationChainIndex;
+    // Reward token
+    uint8 rewardTokenIndex;
+    uint48 rewardAmount;
+    // Route token
+    uint8 routeTokenIndex;
+    uint48 routeAmount;
+    // Expiry duration
+    uint24 expiryDuration;
```

Comment:
Is it `expiry` or `duration`? Did you mean `deadline` ?

Reply:
So, to use a smaller number, instead of passing the deadline, the expiry duration is passed to calculate the deadline using `block.timestamp` + `expiryDuration`

Reply:
What do you think about `expiresIn`?

Reply:
Since we have more room in `bytes32` we can change it to 
`uint96 deadline;`

Reply:
Hmmmm I like `expiresIn` better. 👍🏻 

I think we can leave it as a uint24 since that gives a maximum of 194 days to expiry duration, and I'm already using 8 bytes for the salt.

Reply:
Yeah you have so much unused space left in bytes32, what's the point of introducing this limitation?
You can basically have
```
uint64 rewardAmount;
uint64 routeAmount;
uint56 deadline;
bytes8 salt;
```